tracing/src/main/java/com/palantir/tracing/DeferredTracer.java

```diff
@@ -137,21 +136,27 @@ CloseableTrace withTrace() {
         }
 
         if (isObservable && metadata != null && !metadata.isEmpty()) {
-            return new TaggedCloseableTrace(originalTrace, metadata);
+            return new TaggedCloseableTracer(originalTrace, metadata);
         } else {
-            return originalTrace.map(CLOSEABLE_TRACE_FUNCTION).orElse(DefaultCloseableTrace.INSTANCE);
+            return originalTrace.map(CLOSEABLE_TRACE_FUNCTION).orElse(DefaultCloseableTracer.INSTANCE);
         }
```

Comment:
perf nit: to avoid the `Optional` allocation via `map`:

```suggestion
        if (originalTrace.isPresent()) {
            return CLOSEABLE_TRACE_FUNCTION.apply(originalTrace.get());
        }
        return DefaultCloseableTracer.INSTANCE;
```

Reply:
I've pushed an additional refactor to clean this up more.